src/Views/CountDownView.vala

```diff
@@ -119,6 +125,9 @@ public class CountDownView : Gtk.Box {
 
     private void pause_countdown () {
         if (!is_paused) {
+            // Immediately stop the countdown Timeout - This avoids unnecessary callback
+            Source.remove (countdown);
```

Comment:
I can't reproduce any GLib-CRITICAL with this one: the Timeout should always be active when you press the pause button

Reply:
Ah, I was a fool, it should always be active because clicking cancel button discards the current view and we are impossible to click the pause button.